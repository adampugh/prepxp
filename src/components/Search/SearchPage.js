import React, { Component } from "react";

import Navbar from "../UI/navbarLoggedIn";
import Footer from "../UI/footer";
import SearchBanner from "./searchBanner";
import SearchGrid from "./searchGrid";

class SearchPage extends Component {
    state = {
        filteredLists: [
            {
                title: "Angular Developer",
                questions: [1, 2, 3, 4]
            }, {
                title: "Node JS Developer",
                questions: [1, 2, 5]
            }, {
                title: "General Interview Questions 1",
                questions: [1,1,1,1,1,1,1]
            }
        ]
    }


    render() {
        return (
            <div>
                <Navbar />
                <SearchBanner />
                <SearchGrid filteredLists={this.state.filteredLists}/>
                <Footer />
            </div>
        )
    }
}

export default SearchPage;