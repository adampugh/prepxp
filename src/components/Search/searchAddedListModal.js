import React from "react";
import Modal from "react-modal";

import Loader from "../UI/loader";

const customStyles = {
    overlay : {
        zIndex: 2000,
        background: "rgba(0, 0, 0, 0.6)"
      }
}

const searchAddedListModal = (props) => (
    <Modal
        isOpen={props.isOpen}
        appElement={document.getElementById('root')} 
        onRequestClose={props.closeModal}
        style={customStyles}
        closeTimeoutMS={200}
        className="modal"
    >
        <div className="modal--delete">
            <span onClick={props.closeModal}><i className="fas fa-times"></i></span>
            {props.loading 
                ? <Loader />
                : (
                    <div>
                        <h2>Added list</h2>
                        <p className="modal__icon"><i className="fas fa-check"></i></p>
                    </div>
                )
            }
            
        </div>
    </Modal>
);

export default searchAddedListModal;