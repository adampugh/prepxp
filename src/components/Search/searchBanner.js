import React, { Component } from "react";

class SearchBanner extends Component {
    handleOnChange = (e) => {
        this.props.filterByText(e.target.value.toLowerCase());
    }


    render() {
        return (
            <div className="searchBanner__wrapper">
                <div className="container-s searchBanner__content">
                    <h1>Search lists</h1>
                    <div className="searchBanner__content__searchBar">
                        <input type="text" placeholder="Back-End Developer" onChange={(e) => this.handleOnChange(e)}/>
                        <button className="btn"><i className="fas fa-search"></i> Search  </button>
                    </div>
                </div>
                <div className="searchBanner__whiteElipse"></div>
            </div>
        )
    }
}

export default SearchBanner;
