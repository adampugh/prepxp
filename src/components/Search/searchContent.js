import React from "react";

import SearchSidebar from "./searchSidebar";
import SearchGrid from "./searchGrid";

const searchContent = (props) => (
    <div className="searchContent">
        <div className="container-l">
            <div className="searchContent__grid">
                <SearchSidebar 
                    searchLists={props.searchLists}
                    filterByTag={props.filterByTag} 
                    clearFilter={props.clearFilter} />
                <SearchGrid 
                    searchLists={props.filteredList}
                    handleAddSearchList={props.handleAddSearchList} />
            </div>
        </div>
    </div>
);

export default searchContent;