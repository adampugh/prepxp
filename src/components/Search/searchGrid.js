import React from "react";


const searchGrid = (props) => (
    <div className="searchGrid__lists container-l">
        {props.searchLists.length > 0 ? props.searchLists.map((list, index) => (
            <div className="searchGrid__list__item" key={`${list.title + index}`}>
                <div>    
                    <h2>{list.title}</h2>
                    <p>{Object.keys(list.list.questions).length} Questions</p>
                </div>
                <h2 
                    className="searchGrid__list__icon" 
                    onClick={() => props.handleAddSearchList(list.list)}>
                    <i className="fas fa-plus"></i>
                </h2>
            </div>
        )) : <h2 className="searchGrid__empty"> Your search did not return any results</h2>}
    </div>
);

export default searchGrid;