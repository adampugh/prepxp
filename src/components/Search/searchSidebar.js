import React, { Component } from "react";

class SearchSidebar extends Component {
    state = {
        numOfLists: {},
        tagsArr: []
    }

    componentWillMount() {
        this.getNumOfLists();     
    }

    getNumOfLists = () => {
        // creates object with list totals by sector
        let numOfLists = {}
        this.props.searchLists.forEach(list => {
            numOfLists[list.tag] === undefined 
                ? numOfLists[list.tag] = 1
                : numOfLists[list.tag] += 1;
        });
        this.setState({
            numOfLists
        })
        this.getListTags(numOfLists);
    }

    getListTags = (obj) => {
        let tagsArr = [];
        for (let key in obj) {
            tagsArr.push(key);
        }
        this.setState({
            tagsArr
        });
    }

    render() {
        let icons = {
            Tech: "fas fa-mobile-alt",
            Sales: "fas fa-suitcase",
            Construction: "fas fa-building",
            Transport: "fas fa-truck",
            HR: "fas fa-users",
            Finance: "fas fa-dollar-sign",
            General: "fas fa-plus"
        }

        return (
            <div className="search__sidebar__container">
                <div className="search__sidebar">
                    <h1>Sector</h1>
                    <hr />
                    {this.state.tagsArr.map(tag => (
                        <div key={tag} className="search__sidebar__item" onClick={() => this.props.filterByTag(tag)}>
                            <h2><i className={icons[tag]}></i> {tag}</h2><p>{`(${this.state.numOfLists[tag]})`}</p>
                        </div>
                    ))}
                    <hr />
                    <div className="search__sidebar__item" onClick={this.props.clearFilter}>
                        <h2><i class="fas fa-times"></i> Clear filter</h2>
                    </div>
                </div>
            </div>
        )
    }
}

export default SearchSidebar;



// <div className="search__sidebar__item">
// <h2>General</h2><p>{`(${this.state.numOfLists.general})`}</p>
// </div>
// <div className="search__sidebar__item">
// <h2>Tech</h2><p>{`(${this.state.numOfLists.tech})`}</p>
// </div>
// <div className="search__sidebar__item">
// <h2>Sales</h2><p>{`(${this.state.numOfLists.sales})`}</p>
// </div>
// <div className="search__sidebar__item">
// <h2>Construction</h2><p>{`(${this.state.numOfLists.construction})`}</p>
// </div>
// <div className="search__sidebar__item">
// <h2>Human Resources</h2><p>{`(${this.state.numOfLists.humanResources})`}</p>
// </div>
// <div className="search__sidebar__item">
// <h2>Transport</h2><p>{`(${this.state.numOfLists.transport})`}</p>
// </div>