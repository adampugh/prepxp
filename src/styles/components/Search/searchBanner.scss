.searchBanner__wrapper {
    background: $purple-gradient, url("../assets/images/bg5.jpg") no-repeat center; 
    background-size: cover; 
    text-align: center;
    padding: 150px 0 $padding-m 0;
    color: white;
    position: relative;
    min-height: 180px;
    h1 {
        @include header;
        margin-bottom: 10px;
    }
    input {
        padding: 12px;
        border-radius: 10px;
        border: none;
        box-sizing: border-box;
        width: 100%;
        font-size: $font-s;
        @include boxshadow;
    }
    button {
        margin-top: $padding-s;
    }
    .searchBanner__content__searchBar {
        display: grid;
        grid-template-columns: 1fr;
        z-index: 100;
        input, button {
            z-index: 100;
        }
    }
}

.searchBanner__whiteEllipse {
    -webkit-clip-path: ellipse(73% 87% at 68% 94%);
    clip-path: ellipse(73% 87% at 68% 94%);
    background: $very-light-grey;
    height: 100px;
    position: absolute;
    width: 100%;
    bottom: 0;
}


@media (min-width: $desktop-breakpoint) {
    .searchBanner__wrapper {
        input {
            width: 95%;
        }
        button {
            margin-top: 0;
        }
        .searchBanner__content__searchBar {
            display: grid;
            grid-template-columns: 3fr 1fr;
        }
    }
    
}