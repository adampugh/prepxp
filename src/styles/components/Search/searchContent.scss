.searchContent {
    background: $very-light-grey;
}

.searchContent__grid {
    display: grid;
    grid-template-columns: 1fr;
    background: $very-light-grey;
    padding-bottom: $padding-s;
    min-height: 100vh;
}

@media (min-width: $desktop-breakpoint) {
    .searchContent__grid {
        grid-template-columns: 2fr 6fr;
    }
}