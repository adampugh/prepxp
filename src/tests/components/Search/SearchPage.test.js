import React from "react";
import { shallow } from "enzyme";

import { SearchPage } from "../../../components/Search/SearchPage";

describe("<SearchPage />", () => {
    it("should match snapshot", () => {
        const wrapper = shallow(<SearchPage />);
        expect(wrapper).toMatchSnapshot();
    });
});