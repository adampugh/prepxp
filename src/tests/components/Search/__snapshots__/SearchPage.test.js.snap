// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchPage /> should match snapshot 1`] = `
<div>
  <searchAddedListModal
    closeModal={[Function]}
    isOpen={false}
    loading={true}
  />
  <Navbar />
  <SearchBanner
    filterByText={[Function]}
  />
  <searchContent
    clearFilter={[Function]}
    filterByTag={[Function]}
    handleAddSearchList={[Function]}
  />
  <footer />
</div>
`;
